app/reducers/notificationsFeed.js

```diff
@@ -20,6 +20,8 @@ export default function notificationsFeed(
         ...state,
         ...action.payload
       };
+    case NotificationsFeed.MARK_ALL.SUCCESS:
+      return initialState;
```

Comment:
:+1:

Reply:
Wouldn't you want the red background color from the unread events to disappear as well? While testing this locally it seemed like the counter got cleared, but the background color didn't disappear until the next server fetch.

Maybe you want to map over the notifications in `reducers/notifications.js` and set them all to read? Something like:

```js
    case Notifications.MARK_ALL.SUCCESS:
      return {
        ...state,
        items: state.items.map(item => ({ ...item, read: true }))
      };
```

Reply:
I don't think that's what you would want, no? If so there would be no way to tell which notifications are new. I think it might be unintuitive if your notification bells tells you you have three new notifications, but when you open the dropdown you see a list of all notifications you've ever gotten that all look alike. Removing the color on the next server fetch seems as good a way as any to determine when a notification shouldn't be "new" anymore.

Reply:
I thought we were marking them as read when the notification box is closed - not when we open it? I agree that they should be read when you open, but if they're marked as read when you close the box that would work right?

Reply:
Hm, removing the color when the box is closed could work. Facebook might not be the ideal to strive for, but just for reference they keep notifications marked with a different background color for a while even after you open and close the notification list. In any case, I believe `MARK_ALL_SUCCESS` happens when the list is opened, not closed, as this is what happens when the list is opened or closed:

```
<Dropdown
        show={this.state.notificationsOpen}
        toggle={() =>
          this.setState(
            {
              notificationsOpen: !this.state.notificationsOpen
            },
            () =>
              this.state.notificationsOpen
                ? fetchNotifications()
                : this.props.markAllNotifications()
          )}
```

So if we want to reset the background color after closing the dropdown we could look at the reducer that handles `fetchNotifications()` or some such. I don't really have any strong feelings for whether that's the desired functionality or not.

Reply:
The notifications are marked as read when closing the box. It was intentional to keep the background color even though the counter is cleared, but I can see that it might make sense to make the color disappear. I also see that if we decide to only make the counter disappear, I can make the code a bit cleaner by calling `markAllNotificaitons` when the box opens, then I won't need to check `badgeCount={this.state.notificationsOpen ? 0 : unreadCount}`